include/nbl/builtin/hlsl/mpl.hlsl

```diff
@@ -69,47 +54,72 @@ uint64_t asuint64_t<float64_t>(float64_t val)
 }
 #endif
 
-template<uint64_t N>
+namespace impl
+{
+
+template<uint64_t N, uint16_t bits>
 struct countl_zero
 {
-    NBL_CONSTEXPR_STATIC_INLINE uint16_t value =
-#ifdef __HLSL_VERSION
-      impl::countl_zero<N, 6>::value;
+    NBL_CONSTEXPR_STATIC_INLINE uint64_t SHIFT = bits >> 1;
+    NBL_CONSTEXPR_STATIC_INLINE uint64_t LO_MASK = (1ull << SHIFT) - 1;
+    NBL_CONSTEXPR_STATIC_INLINE bool CHOOSE_HIGH = N & (LO_MASK << SHIFT);
+    NBL_CONSTEXPR_STATIC_INLINE uint64_t NEXT = (CHOOSE_HIGH ? (N >> SHIFT) : N) & LO_MASK;
+    NBL_CONSTEXPR_STATIC_INLINE uint16_t value = countl_zero<NEXT, SHIFT>::value + (CHOOSE_HIGH ? 0ull : SHIFT);
+};
+
+template<uint64_t N>
+struct countl_zero<N, 1> : integral_constant<uint16_t, uint16_t(1u - (N & 1))>
+{};
+
+}
+
+template<class T, T N>
+struct countl_zero : impl::countl_zero<uint64_t(N), (sizeof(T) * 8)>
+{
+#ifdef __cplusplus
+    static_assert
 #else
-      std::countl_zero(N);
+    _Static_assert
 #endif
```

Comment:
just make an `#define static_assert(...) _Static_assert(__VA_ARGS__)` in `macros.h` for the __HLSL_VERSION path